File: 728._Self_Dividing_Numbers.py
class Solution:
    def check(self,num):
        temp = num
        while temp:
            if not temp % 10 or num % (temp % 10): return False
            temp = temp//10
        return True
    def selfDividingNumbers(self, left, right):
        result=[]
        for i in range(left,right+1):
            if(self.check(i)):
                result.append(i)
        return result
        

File: 728._Self_Dividing_Numbers.cpp
class Solution {
public:
    bool check(int n){
        for(int i=n;i>0;i/=10){
            if(!(i%10) || n % (i % 10))
                return false;
        }
        return true;
        
    }
    
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        for(int i=left;i<=right;i++){
            if(check(i)){
                result.push_back(i);
            }
        }
        return result;
    }

};